api/nodes/views.py

```diff
@@ -1149,7 +1159,17 @@ def get_queryset(self):
         if not hasattr(node_addon, 'get_folders'):
             raise EndpointNotImplementedError('Endpoint not yet implemented for this addon')
 
-        return node_addon.get_folders(path=path, folder_id=folder_id)
+        #  Convert v1 errors to v2 as much as possible.
+        try:
+            return node_addon.get_folders(path=path, folder_id=folder_id)
+        except InvalidAuthError:
+            raise NotAuthenticated
+        except HTTPError as exc:
+            v2_exception = HTTP_CODE_MAP.get(exc.code)
+            if v2_exception:
+                raise v2_exception
+            else:
+                raise exc
```

Comment:
Wondering if we should throw a DRF exception here, even it's another 500 error, as a catchall.  Not sure what the most appropriate thing is, but here's a thought:

```
class ProviderError(APIException):
    status_code = status.HTTP_500_INTERNAL_SERVER_ERROR
    default_detail = _('Unknown provider error.')
```

<img width="457" alt="screen shot 2018-06-21 at 3 46 18 pm" src="https://user-images.githubusercontent.com/9755598/41744748-5471c76c-756a-11e8-9ca8-4abcb441b143.png">